### DISCUSSION TITLE
Have you heard of lunatic?

### DISCUSSION THREAD

**Discussion by sid-707** [linkdd/letlang#3] [General]
Hey David,

Letlang is a really interesting and ambitious project. I especially like the documentation on the website and in your medium articles.

Have you heard of [lunatic](https://github.com/lunatic-solutions/lunatic) by any chance?

It's a WebAssembly runtime inspired by Erlang written in Rust. It even uses `tokio`.

I see a lot of overlapping goals between letlang and lunatic and thought you might be interested in checking it out.

**Comment by linkdd** [comment]
Hello,

Thank you for the feedback ❤️

Yes I've heard of lunatic, I even considered using it in the Letlang Runtime and replace my current implementation (assuming you can run WebAssembly outside of a browser, I admit to know next to nothing about WASM). Though that's for later, I need to finish the language's spec first before continuing the implementation.

I'm at the "Proof of Concept" phase and progressing slowly (I wish I could work fulltime on this, but I need food 😅).

**Reply by sid-707** [reply]
That's awesome 😃 

Yes WebAssembly [runs outside the browser](https://youtu.be/CMB6AlE1QuI?t=2112). Imo though WebAssembly is just the cherry on top.

I think making Letlang compile to Rust is very interesting and puts it in a very fortuitous position to run on lunatic. There's a lunatic library to integrate with Rust: [lunatic-rs](https://github.com/lunatic-solutions/lunatic-rs). I'll play around with it and Letlang to see if I can build a POC to integrate the two. If it does, there are two advantages: 

1. Letlang could become just a language and not a runtime. Like you said, this way you can replace your current implementation. It reduces complexity and you can just focus on language design.
2. Letlang gets access to lunatic's ecosystem such as [submillisecond](https://github.com/lunatic-solutions/submillisecond) which is a web framework in the works.

The reason why I'm interested in this is because I was looking to see if I could build a source-to-source compiler (I'm inspired by Typescript too 😄). My goal is to write code in a functional/ immutable way that compiles to Rust.

Are you on Discord?

**Reply by linkdd** [reply]
Sorry for the late reply, I did not see your message.

Yes I'm on discord, and I made a server to talk about some of my projects, though it's pretty empty for now.

https://discord.gg/QmPZSAJtWm

**Comment by WarnerMarkakisA** [comment]
'm glad to hear that you find the project interesting and that you appreciate the documentation.

And yes, I have heard of Lunatic! It's definitely an interesting project, and I can see why you would draw parallels between it and Letlang. Both projects aim to bring some of the strengths of Erlang to a new platform - in the case of Letlang, that platform is the web, and in the case of Lunatic, it's WebAssembly.

I haven't had a chance to play around with Lunatic too much yet, but from what I've seen, it looks like a really promising project. It's always exciting to see new technologies that take inspiration from Erlang and the BEAM, and I'm looking forward to seeing how Lunatic develops over time.

**Reply by linkdd** [reply]
Once you get a taste of the OTP framework, you can never go back 🙂
